src/district0x/ui/events.cljs

```diff
@@ -796,3 +799,18 @@
   interceptors
   (fn [_ [events & args]]
     {:dispatch-n (mapv #(vec (concat % args)) (remove nil? events))}))
+
+(reg-event-fx
+ :district0x/reload-address
+ interceptors
+ (fn [{:keys [db]}]
+   {:dispatch-n [[:district0x/load-my-addresses]
```

Comment:
When you `load-my-addresses` you must actually wait until it's loaded and then start watching, otherwise might watch different addresses. 
```
:async-flow {:first-dispatch [:district0x/load-my-addresses]
              :rules [{:when :seen?
              :events [:district0x/my-addresses-loaded]
              :dispatch [:district0x/watch-my-eth-balances]}]}
```